### PR TITLE
Add docs for create new extension layer #14

### PR DESCRIPTION
**Pull request by sabmeua**
Support for #14

### CODE REVIEW

File: docs/create_your_own_extension_layer.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+```Dockerfile
+
+WORKDIR ${PHP_BUILD_DIR}/ext/pgsql
+RUN phpize
+RUN ./configure --with-pgsql=${INSTALL_DIR}
+RUN make -j `nproc` && make install
+
+RUN cp `php-config --extension-dir`/pgsql.so /tmp/pgsql.so
+```
+Some env variables are set and can be used in this environment. `PHP_BUILD_DIR` is `/tmp/build/php`, `INSTALL_DIR` is `/opt/bref`.
```
Comment (Nyholm) [inline docs/create_your_own_extension_layer.md]: We should mention that we get the environment variables from `bref/build-php-$PHP_VERSION`. That will be great if you need to find out what more is available.

File: docs/create_your_own_extension_layer.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+```Dockerfile
+COPY --from=ext /tmp/pgsql.so /opt/bref-extra/pgsql.so
+```
+
+Extension files that need to be installed should be placed `/opt` directory in the final image.
```
Comment (Nyholm) [inline docs/create_your_own_extension_layer.md]: We should say why. 

(Lambda only allows us to put things in the `/opt`)

File: docs/create_your_own_extension_layer.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+ env:
+   - LAYER=amqp PHP="72 73 74"
+   - LAYER=blackfire PHP="72 73 74"
+   - LAYER=gmp PHP="72 73 74"
+   - LAYER=memcached PHP="72 73 74"
++  - LAYER=pgsql PHP="72 73 74"
+   - LAYER=xdebug PHP="72 73 74"
+```
+
+Once you have done above please submit the PR.
```
Comment (Nyholm) [inline docs/create_your_own_extension_layer.md]: We should also encourage to submit a PR with "half finish work"

File: docs/create_your_own_extension_layer.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+# Build the final image from the lambci image that is close to the production environment
+FROM lambci/lambda:provided
+
+#
+# Add commands to copy files that requried for final image.
+#
+```
+
+Then you need to add two part of Dockerfile, the following snippets are examples of build and file copy parts respectively.
```
Comment (Nyholm) [inline docs/create_your_own_extension_layer.md]: `s/part/parts/`

File: docs/create_your_own_extension_layer.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+```Dockerfile
```
Comment (Nyholm) [inline docs/create_your_own_extension_layer.md]: Maybe introduce this part of the dockerfile a little. Say that we need to install the libraries and fix things to eventually install the php extension

File: docs/create_your_own_extension_layer.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+RUN phpize
+RUN ./configure --with-pgsql=${INSTALL_DIR}
+RUN make -j `nproc` && make install
+
+RUN cp `php-config --extension-dir`/pgsql.so /tmp/pgsql.so
+```
+Some env variables are set and can be used in this environment. `PHP_BUILD_DIR` is `/tmp/build/php`, `INSTALL_DIR` is `/opt/bref`.
+
+```Dockerfile
+COPY --from=ext /tmp/pgsql.so /opt/bref-extra/pgsql.so
```
Comment (Nyholm) [inline docs/create_your_own_extension_layer.md]: We should also explain this a little. Ie, we need to copy every library that our extension is using.

**Comment by sabmeua** [comment]
Thanks! I changed descriptions a little. Could you check it out?

File: docs/create_your_own_extension_layer.md [hunk 7]
```diff
@@ -38,13 +38,22 @@ RUN make -j `nproc` && make install
 
 RUN cp `php-config --extension-dir`/pgsql.so /tmp/pgsql.so
 ```
-Some env variables are set and can be used in this environment. `PHP_BUILD_DIR` is `/tmp/build/php`, `INSTALL_DIR` is `/opt/bref`.
+
+Compile the extension here.
+In some cases, download the source code, install the libraries required for compilation, perform pecl install, etc, describe the processing to create all files required for extension.
```
Comment (Nyholm) [inline docs/create_your_own_extension_layer.md]: > describe the processing to create all files required for extension.

Is this a comment to yourself? =)

Comment (sabmeua) [reply docs/create_your_own_extension_layer.md]: hmm.. What I mean is that all necessary processing needs to be written here.
I'm sorry my English isn't very good.

Comment (sabmeua) [reply docs/create_your_own_extension_layer.md]: I removed that sentence. How is this? :)

**Comment by Nyholm** [comment]
Thank you

**Comment by Nyholm** [comment]
I added some minor changes. I think this is a good first version

**Pull request merged**